### PR TITLE
[FIX] portal: add ellipses to search box placeholder overflow 

### PR DESCRIPTION
**Pull request by lvsz**
Versions:
---------
- 15.0+

Issue:
------
Depending on the font, string & window size, the `oe_search_box` placeholder may get cropped awkwardly.

Example:
example

Cause:
------
The search field can't guarantee adequate width to show the full placeholder string.

Solution:
---------
Add `text-overflow: ellipsis` CSS rule for `oe_search_box` elements to make it clear the placeholder was cropped.

Related:
--------
- https://github.com/odoo/enterprise/pull/47671
- https://github.com/odoo/enterprise/pull/48973

opw-3504281

### CODE REVIEW

**Comment by lvsz** [comment]
Before:
appoint_before
After:
appoint_after

File: addons/portal/static/src/scss/portal.scss [hunk 1]
```diff
@@ -205,6 +205,9 @@ img, .media_iframe_video, .o_image {
 .oe_search_box {
     padding-right: 23px;
 }
+.oe_search_box::placeholder {
+    overflow: visible;
```
Comment (thiryr) [inline addons/portal/static/src/scss/portal.scss:209]: Mmm, here it happens to fit but what if the text is even longer?

Might be better to set `text-overflow: ellipsis` on the search_box instead so we're sure not to introduce additional weirdness in here, if that seems ok to you

**Review by rdeodoo — CHANGES_REQUESTED** [review]
You can very well fix this specific one, by considering what @thiryr said, but I would not do anything, except maybe rewording the "Search an appointment.." to just "Search" in master.
That's what we do on blog, forum, ecommerce search input in order to solve this issue.

Keep in mind that English words are almost always tiny, compared to their translated version, especially in French, German..
So, when fixing such "wording space" issues, a good rule of thumbs is to consider that the current text **will** be twice longer in other languages, and to find a solution that would make it work in that case too (maybe even 3 times longer if you consider the extreme cases to be sure).

I am also sure there is other places that suffer the same issue, so fixing this specific one is nice but not mandatory IMHO.

You could just tell the customer that it's not ideal, that we will think about it in the next Odoo version, that he (the client) was heard but that we won't change it in stable versions.

Just my 2 cents passing by, feel free to fix this specific case ;-)

**Comment by lvsz** [comment]
> You can very well fix this specific one, by considering what @thiryr said, but I would not do anything, except maybe rewording the "Search an appointment.." to just "Search" in master.
> That's what we do on blog, forum, ecommerce search input in order to solve this issue.

That is a good point. Given that it's located on the appointments page, having "Search an appointment" doesn't provide any extra value over simply "Search"

Closing this PR and opening one in Enterprise instead https://github.com/odoo/enterprise/pull/47671

File: addons/portal/static/src/scss/portal.scss (hunk 1 above)
Comment (lvsz) [reply addons/portal/static/src/scss/portal.scss:209]: Interesting, didn't know about that specific property yet. It's more appropriate than the one I used, but I'm going with [rdeodoo's](https://github.com/odoo/odoo/pull/135801#pullrequestreview-1635235635) suggestion instead of changing the placeholder string itself.

**Comment by rdeodoo** [comment]
Could you please make sure the runbot is green so we can review / ship this one?

**Comment by lvsz** [comment]
@rdeodoo It got killed for some reason, seems like a simple rebuild did the trick, should be good now.

**Comment by lvsz** [comment]
Updated PR & commit message to reflect recent changes.

**Comment by lvsz** [comment]
@rdeodoo

**Comment by lvsz** [comment]
@robodoo retry
